diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter2/comments/CommentsRule.kt

```diff
@@ -85,7 +86,7 @@ class CommentsRule(configRules: List<RulesConfig>) : DiktatRule(
             }
             .forEach { (offset, parsedNode) ->
                 COMMENTED_OUT_CODE.warn(configRules, emitWarn, isFixMode, parsedNode.text.substringBefore("\n").trim(), offset,
-                    errorNodesWithText.find { it.second == parsedNode.text }?.first ?: parsedNode)
+                    errorNodesWithText.find { it.second.trim().contains(parsedNode.text, false) || parsedNode.text.contains(it.second.trim(), false) }?.first!!)
```

Comment:
Why are you sure, that there will be a matching element? Why did we have `?: parsedNode` here?

Reply:
parsedNode is a custom node we created. We need to find the parent node of the code text on which we throw the warning.

Reply:
I mean, we can guarantee, that the element will always be found? Because of logic of `errorNodesWithText` collection?

Reply:
Yes, we expect there will always be the right node, because all comment lines are saved in the errorNodesWithText